tests/unit/test_client.py

```diff
@@ -1502,6 +1529,12 @@ def _download_blob_to_file_helper(
             expected_retry = None
 
         headers = {"accept-encoding": "gzip"}
+        _helpers._add_etag_match_headers(
+            headers,
+            if_etag_match=extra_kwargs.get("if_etag_match"),
+            if_etag_not_match=extra_kwargs.get("if_etag_not_match"),
+        )
+
```

Comment:
This would be clearer if we just assembled the values directly, rather than using the helper:

```suggestion
        if_etag_match = extra_kwargs.get("if_etag_match")
        if if_etag_match is not None:
            headers["If-Match"] = if_etag_match
        if_etag_not_match = extra_kwargs.get("if_not_etag_match")
        if if_etag_not_match is not None:
            headers["If-None-Match"] = if_etag_not_match
```

Reply:
Lol, that's originally what I had before I needed to copy/paste the logic into a 3rd place and ended up refactoring out into the helper. Understand the desire for DAMP tests though, so will revert this instance.

Reply:
It is slightly more complex than the suggestion because of str and list.